### DISCUSSION TITLE
Branches for translation on this repository instead of forks

### DISCUSSION THREAD

**Discussion by itsyigid** [saber-notes/saber#264] [General]
Using forks for translation is generally hard since it causes merge conflicts. I've been discarding commits on my fork and recloning it before every translation update to overcome these conflicts.

It goes something like this:
1. Compare adil192:main to fybx:main (my fork)
2. Discard commits on my fork
3. Pull changes from adil192:main
4. Edit `missing_translations.json` and generate new strings file
5. Commit updated files to my fork
6. Compare and open pull request to base (adil192:main) from my fork
7. @adil192 accepts my pull request and merges my branch
8. At this point, my fork is again useless, so I need to discard commits on my fork

Using a dedicated branch for translations like `adil192:translation-tr` and keeping this branch up-to-date only for translation updates would be much simpler. When a new commit is on the translation branch, opening a pull request from it (`adil192:translation-XX`) to the main branch (`adil192:main`) would be sufficient and straightforward.

**Comment by adil192** [comment]
> Using forks for translation is generally hard since it causes merge conflicts. I've been discarding commits on my fork and recloning it before every translation update to overcome these conflicts.

If I'm not mistaken, the merge conflicts occur because each i18n pull request is squashed into the `adil192:main` branch. 

> Using a dedicated branch for translations like `adil192:translation-tr` and keeping this branch up-to-date only for translation updates would be much simpler. When a new commit is on the translation branch, opening a pull request from it (`adil192:translation-XX`) to the main branch (`adil192:main`) would be sufficient and straightforward.

This seems overly complicated and I'm not sure it would make things easier.

***

There are 2 ways to have a simpler process than the 8-step one you described:

1. Create a new branch on your fork each time you want to edit `_missing_translations.json`. This keeps your `main` branch in sync with `adil192:main` so you can run `git pull` with no conflicts. 

2. Use GitHub's web-based editor. It will create a new branch automatically for you (e.g. `patch-1`). This is the method I outline in https://github.com/adil192/saber#extending-existing-languages.
